### PR TITLE
Add an extended source spectral analysis tutorial

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request adds a tutorial explaining how to perform extended source analysis in gammapy. It uses RX J1713 data. It runs in 9sec on my machine. It follows the description introduced some time ago in PR #2672. 

Maybe some discussion could be added on the question of handling of observations with no OFF regions because it is likely to be more frequent for extended objects.

**Dear reviewer**

This should be ready for review.

### CODE REVIEW

**Comment by adonath** [comment]
Thanks @registerrier, I agree this is a very useful tutorial (I already know that a few users wanted to do such an analysis...).

I'll do a more detailed review this afternoon, but for now I have one early comment:

I guess in practice it might be hard to find any reflected regions for extended sources in the Galactic Plane (this was exactly the problem we had for the HGPS)? So one could show a slightly different approach in addition:
- Extract a 3D MapDataset
- Use FoVBackgroundEstimator 
- Create a `SpectrumDataset` with fixed background
- Do the fit and compare

What do you think?

**Comment by registerrier** [comment]
It is easy indeed to modify the notebook to perform a fit with the `SpectrumDataset`. If you simply add `"background"` to the `MapDatsetMaker` init and  comment the line calling the `bkg_maker.run(dataset,obs)` you have a fit with a fixed background model. And it works. The result is rather good, but this is not really surprising given the brightness of this source.

We could indeed advertise other ways to perform fits that are made possible with gammapy but we probably should test them a bit or ask people to test these ideas.

File: tutorials/extended_source_spectral_analysis.ipynb [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+    "We first set the datastore and retrieve a few observations from our source."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datastore_path = \"/Users/terrier/Code/gammapy-dev/gammapy-datasets\"\n",
```
Comment (bkhelifi) [inline tutorials/extended_source_spectral_analysis.ipynb]: The path should be modified to maybe $GAMMAPY_DATA

Comment (registerrier) [reply tutorials/extended_source_spectral_analysis.ipynb]: Indeed. Thanks.

**Comment by adonath** [comment]
@bkhelifi With the current implementation of the `Analysis` class we will only have a limited selection of analysis workflows available. So we cannot attach the corresponding YAML files to the notebooks, because those (special) analysis workflows are not possible using the HLI (and as long as we have the hard-coded workflows, they will never be IMHO...the options of combining makers and create custom data reduction chains are just endless...sow we stick to the 2nd level API for that).

**Comment by registerrier** [comment]
Travis CI fail is unrelated. Merging.

I propose to move the discussion about what to have as tutorials in HLI vs mid-level API to an issue.

**Pull request merged**